### PR TITLE
Retry policies should clone HTTP requests

### PR DESCRIPTION
**Pull request by daschult**

### CODE REVIEW

File: lib/policies/systemErrorRetryPolicy.ts [hunk 1]
```diff
@@ -115,10 +115,10 @@ export class SystemErrorRetryPolicy extends BaseRequestPolicy {
       // If previous operation ended with an error and the policy allows a retry, do that
       try {
         await utils.delay(retryData.retryInterval);
-        const res: HttpOperationResponse = await this._nextPolicy.sendRequest(operationResponse.request);
-        return self.retry(res, retryData, err);
+        const res: HttpOperationResponse = await this._nextPolicy.sendRequest(request.clone());
```
Comment (RikkiGibson) [inline lib/policies/systemErrorRetryPolicy.ts]: Is this a redundant clone? (also cloning in sendRequest)

File: lib/webResource.ts [hunk 2]
```diff
@@ -231,6 +231,18 @@ export class WebResource {
...
+
+  /**
+   * Clone this WebResource HTTP request object.
+   * @returns {WebResource} The clone of this WebResource HTTP request object.
+   */
+  clone(): WebResource {
+    const result = new WebResource(this.url, this.method, this.body, this.query, this.headers, this.rawResponse, this.abortSignal);
+    result.formData = this.formData;
+    result.operationSpec = this.operationSpec;
+    result.abortSignal = this.abortSignal;
```
Comment (RikkiGibson) [inline lib/webResource.ts]: redundant?

File: lib/policies/systemErrorRetryPolicy.ts (hunk 1 above)
Comment (daschult) [reply lib/policies/systemErrorRetryPolicy.ts]: No, we need to clone for each attempt to send the request.

Comment (RikkiGibson) [reply lib/policies/systemErrorRetryPolicy.ts]: Oh, you're right. I see the request passed to sendRequest on 118 is completely independent from the retry.

File: lib/webResource.ts (hunk 2 above)
Comment (daschult) [reply lib/webResource.ts]: Removed in https://github.com/Azure/ms-rest-js/pull/100/commits/5fd6b70d34d2521918ec6630e02a13288f41c69f.

**Pull request merged**